docs/src/references/tedge-write.md

```diff
@@ -0,0 +1,73 @@
+---
+title: tedge-write
+tags: [Reference, Configuration, CLI]
+description: Description of tedge-write component
+---
+
+**tedge-write** is a helper thin-edge component used by tedge-agent for privilege elevation.
+tedge-agent spawns a `tedge-write` process when it needs to write to files that `tedge` user/group
+has no write permissions to (e.g. system files or files owned by other packages), for example when
+writing an updated configuration file as part of [`config_update` operation][1]. `tedge-agent` will
+first try to write to a file directly and only retry using `tedge-write` if direct write fails due
+to `tedge` user/group not having write permissions to either the file itself or its parent
+directory. Write permission to the parent directory is required to guarantee config updates are
+atomic.
+
+[1]: agent/tedge-configuration-management.md#handling-config-update-commands
+
+`tedge-write` will perform an atomic write, i.e. it will write to a temporary file, set file
+ownership and mode, and finally rename the temporary into the final file. If the target file already
+exists, its original ownership/mode will be preserved and optionally provided new values will be
+ignored.
```

Comment:
I would organize the doc a bit differently:

- First, the aim of `tedge-write` and when it's used.
- Second, how to configure `sudo` to grant write access to specific files.
- And only then, tell how `tedge-write` works and what it does.

Reply:
fixed 6ed9c1c9b7